src/ansi-c/c_typecheck_initializer.cpp

```diff
@@ -120,13 +121,16 @@ exprt c_typecheck_baset::do_initializer_rec(
       {
         // fill up
         tmp.type()=type;
-        const auto zero =
-          zero_initializer(full_type.subtype(), value.source_location(), *this);
+        const auto zero = zero_initializer(
+          to_array_type(full_type).element_type(),
+          value.source_location(),
+          *this);
```

Comment:
Several lines above we should already have introduced something like `const array_typet &array_type = to_array_type(full_type);`.

Reply:
done